nixos/modules/hardware/onlykey.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, ... }:
+
+with lib;
+
+{
+
+  ####### interface
+
+  options = {
+
+    hardware.onlykey = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable OnlyKey device (https://crp.to/p/) support.
+        '';
+      };
+    };
+
+  };
+
+  ## As per OnlyKey's documentation piece (hhttps://docs.google.com/document/d/1Go_Rs218fKUx-j_JKhddbSVTqY6P0vQO831t2MKCJC8),
+  ## it is important to add udev rule for OnlyKey for it to work on Linux
+
+  ####### implementation
+
+  config = mkIf config.hardware.onlykey.enable {
+    services.udev.extraRules =
+      ''
+      # UDEV Rules for Teensy boards, http://www.pjrc.com/teensy/
+      #
+      # The latest version of this file may be found at:
+      #   http://www.pjrc.com/teensy/49-teensy.rules
+      #
+      # This file must be placed at:
+      #
+      # /etc/udev/rules.d/49-teensy.rules    (preferred location)
+      #   or
+      # /lib/udev/rules.d/49-teensy.rules    (req'd on some broken systems)
+      #
+      # To install, type this command in a terminal:
+      #   sudo cp 49-teensy.rules /etc/udev/rules.d/49-teensy.rules
+      #
+      # Or use the alternate way (from this forum message) to download and install:
+      #   https://forum.pjrc.com/threads/45595?p=150445&viewfull=1#post150445
+      #
+      # After this file is installed, physically unplug and reconnect Teensy.
+      #
+      ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04[789B]?", ENV{ID_MM_DEVICE_IGNORE}="1"
+      ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04[789A]?", ENV{MTP_NO_PROBE}="1"
+      SUBSYSTEMS=="usb", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="04[789ABCD]?", MODE:="0666"
```

Comment:
Mode "0666" is questionable. That adds RW access to all users.

Typically udev rules add RW access to all users in the "plugdev" group. Which is users who are authorized to "plug in devices".

Reply:
I think the new style is to use ```TAG+="uaccess"``` instead of using GROUP= and MODE=.

Reply:
@bjornfor I just tested with `TAG+="uaccess"` on my local system and the device didn't work, switch back to `GROUP` made it work. Not yet sure why.

Reply:
Looking at other packages it looks like that'd require creating a new udev file with an earlier precedence. Is this what you want me to do?